users/postgres/users.go

```diff
@@ -290,17 +366,30 @@ func toUser(dbu dbUser) (users.User, error) {
 	}, nil
 }
 
-func getUsersMetadataQuery(m users.Metadata) ([]byte, string, error) {
-	mq := ""
-	mb := []byte("{}")
-	if len(m) > 0 {
-		mq = ` AND users.metadata @> :metadata`
+func createEmailQuery(entity string, email string) (string, string, error) {
+	query := ""
+	param := ""
+
+	if email != "" {
+		// Create LIKE operator to search Users with email containing a given string
+		param = fmt.Sprintf(`%%%s%%`, email)
+		query = fmt.Sprintf("%semail LIKE :email", entity)
+		return query, param, nil
+	}
+
+	return query, param, nil
+}
 
-		b, err := json.Marshal(m)
+func createMetadataQuery(entity string, meta users.Metadata) (string, []byte, error) {
+	query := ""
+	param := []byte("{}")
```

Comment:
Is param allowed to be `nil` if len(meta) == 0? If so, this can be simplified:
```go

func createMetadataQuery(entity string, meta users.Metadata) (string, []byte, error) {
	if len(meta) == 0 {
		return "", nil, nil
	}
	param, err := json.Marshal(meta)
	if err != nil {
		return "", nil, err
	}
	query := fmt.Sprintf("%smetadata @> :metadata", entity)
	return query, param, nil
}
```